src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -421,12 +428,21 @@ export class ElasticsearchConfig {
 const readKeyAndCerts = (rawConfig: ElasticsearchConfigType) => {
   let key: string | undefined;
   let keyPassphrase: string | undefined;
-  let certificate: string | undefined;
-  let certificateAuthorities: string[] | undefined;
+  let certificate: X509Certificate | undefined;
+  let certificateAuthorities: X509Certificate[] | undefined;
+
+  const validateCertificate = (cert: string, source: string): X509Certificate => {
+    try {
+      return new X509Certificate(Buffer.from(cert));
+    } catch (e) {
+      throw new Error(`Could not read certificate authority from [${source}] due to error: ${e}`);
+    }
+  };
```

Comment:
Here is where the new validation is happening. This is being applied to both CAs and the certificate, both from disk and keystore and truststore